docs/CODE_REVIEW.md

```diff
@@ -0,0 +1,155 @@
+# How to do code reviews for curl
+
+Anyone and everyone is encouraged and welcome to review code submissions in
+curl. This is a guide on what to check for and how to perform a successful
+code review.
+
+## All submissions should get reviewed
+
+All pull requests and patches submitted to the project should be reviewed by
+at least one experienced curl maintainer before that code is accepted and
+merged.
+
+## Let the tools and tests take the first rounds
+
+On initial pull requests, let the tools and tests do their job first and then
+start out by helping the submitter understand the test failures and tool
+alerts.
+
+## How to provide feedback to author
+
+Be nice. Ask questions. Provide examples or suggestions of improvements.
+Assume best intentions. Remember language barriers.
+
+All first-time contributors can become regulars. Let's help them go there.
+
+## Is this a change we want?
+
+If this is not a change that seems to be aligned with the project's path
```

Comment:
unsure if we need a generic 'preflight' checklist when doing a code review ... of which we could highlight/require that every PR needs to reference an issue.

Reply:
What about tidy-up PRs? Would these require an issue?

Reply:
I do not think we need to say ***must*** eg. PR's that do not reference an issue # should be rare.

Reply:
If we should say something about issues, it should then probably be something about the commit message - which the current version of the document skips completely. I don't think we will benefit from requiring an issue to exist before a pull-request is made. Looking back at how we typically work, a rather large portion of our bugfixes are done as a PR right away, shortcutting over the issue if the person who does the fix also is the person who found the issue.

Reply:
code reviewer should encourage contributors to scan existing issue list for previous issues and try to link PR/commit with an issue ... not suggesting we need to backfill issues to satisfy.

Reply:
Agreed, but technically speaking the commit message is not code so it's not part of a "code review". It is however part of the "PR review" or perhaps the things-to-check-before-merge. What do you say about an added section like below?

---

# Commit messages

Tightly coupled with a code review is making sure that the commit message is
good. It is the responsibilitiy of the person who merges the code to make sure
that the commit message follows our standard (detailed in the
[CONTRIBUTE.md](CONTRIBUTE.md) document). This includes making sure the PR
identifies related issues and giving credit to reporters and helpers.